### PR TITLE
tikv: refine commit backoff slow log

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

1. we don't know how much time cause by backoff for a slow commit

https://github.com/pingcap/tidb/blob/f05a27d0bb2af157222840133860dd41d68ebde7/store/tikv/2pc.go#L409

will fork or clone a new backoff to parallel handle key groups and this let backoff time be miss in commit slow log

2. when we have known a commit is slow due to backoff, but we don't know why it backoff 

### What is changed and how it works?

- record delta sleep into parent backoff to let backoff accurate for commit operation
- add backoffTypes fields to commit slow log

after them, we can got log like this:

```
# Time: 2019-08-16T16:32:02.281654197+08:00
# Txn_start_ts: 410502908375465985
# User: root@127.0.0.1
# Conn_ID: 1
# Query_time: 0.132051478
# Prewrite_time: 0.101346878 Commit_time: 0.030348021 Get_commit_ts_time: 0.000101586 Commit_backoff_time: 0.1 Backoff_types: [tikvRPC regionMiss] Local_latch_wait_time: 2.288e-06 Write_keys: 1 Write_size: 23 Prewrite_region: 2
# DB: test
# Is_internal: false
# Digest: db305e24a5f7a0ae2c46a1770efa5150459b121ac3dc1ce39ba5645a4f8521ff
# Num_cop_tasks: 0
# Cop_proc_avg: 0 Cop_proc_p90: 0 Cop_proc_max: 0 Cop_proc_addr: 
# Cop_wait_avg: 0 Cop_wait_p90: 0 Cop_wait_max: 0 Cop_wait_addr: 
# Succ: false
insert into t VALUES(1);
```

this insert is slow due to prewrite is slow and backoff take most time in prewrite.

and backoff type is `Backoff_types: [tikvRPC regionMiss]`

```
Commit_backoff_time: 0.1 Backoff_types: [tikvRPC regionMiss]
```
### Check List 

Tests 

 - Manual test

Code changes

 - log record change

Side effects

 - n/a
Related changes

 - Need to cherry-pick to the release branch

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/11757)

### CODE REVIEW

File: util/execdetails/execdetails_test.go [hunk 1]
```diff
@@ -14,6 +14,9 @@
 package execdetails
 
 import (
+	"fmt"
+	"github.com/pingcap/tidb/util/stringutil"
```
Comment (jackysp) [inline util/execdetails/execdetails_test.go]: format

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -96,7 +97,7 @@ type twoPhaseCommitter struct {
 	// We use it to guarantee GC worker will not influence any active txn. The value
 	// should be less than GC life time.
 	maxTxnTimeUse uint64
-	detail        *execdetails.CommitDetails
+	detail        unsafe.Pointer
```
Comment (crazycs520) [inline store/tikv/2pc.go]: Why change to `unsafe.Pointer`?

Comment (lysu) [reply store/tikv/2pc.go]: in the previous, test was failed due to race condition in https://github.com/pingcap/tidb/pull/11757/files#diff-499c236856cd9ce3300d3f5ccde41a23R441, after this PR detail will be accessed in forked goroutine.

**Comment by jackysp** [comment]
PTAL @crazycs520

File: store/tikv/2pc.go [hunk 3]
```diff
@@ -427,12 +429,22 @@ func (c *twoPhaseCommitter) doActionOnBatches(bo *Backoffer, action twoPhaseComm
...
-				singleBatchBackoffer, singleBatchCancel := backoffer.Fork()
+				var singleBatchCancel context.CancelFunc
+				singleBatchBackoffer, singleBatchCancel = backoffer.Fork()
 				defer singleBatchCancel()
-				ch <- singleBatchActionFunc(singleBatchBackoffer, batch)
+			}
+			beforeSleep := singleBatchBackoffer.totalSleep
+			ch <- singleBatchActionFunc(singleBatchBackoffer, batch)
+			commitDetail := c.getDetail()
+			if commitDetail != nil {
```
Comment (crazycs520) [inline store/tikv/2pc.go]: When `commitDetail `  will be nil?

Comment (lysu) [reply store/tikv/2pc.go]: mainly from pessimistic txn lock:

https://github.com/pingcap/tidb/blob/523b93651d15015c25c8102bb54991f3977ee327/store/tikv/txn.go#L402

https://github.com/pingcap/tidb/blob/523b93651d15015c25c8102bb54991f3977ee327/store/tikv/txn.go#L455

for pessimistic txn, "commit detail" maybe not suitable, maybe better cover pessimistic logic in pessimistic observable task :D

Comment (crazycs520) [reply store/tikv/2pc.go]: Please add a comment here.

**Pull request merged**